### PR TITLE
Introduce "ar" demo module and some test infrastructure

### PR DESCRIPTION
**Pull request by clebergnu**
This introduce a more real demonstration module: `autils/archive/ar.py`, and with it, some of the test infrastructure that is planned (and documented in the [BluePrint](https://avocado-framework.readthedocs.io/en/101.0/blueprints/BP005.html#quality-requirements)).

As it's explained in one individual commit message, one should not consider that the *ar* module in Avocado has been migrated to this repo just yet.  Only when this repository has all the quality checks enforced, and the migration is declared on the Avocado side, one may assume that.

---

This implements https://github.com/avocado-framework/autils/issues/5.

### CODE REVIEW

File: schemas/autils.schema [hunk 1]
```diff
@@ -48,10 +48,15 @@
             ],
             "items": { "type": "string" }
         },
+        "tests": {
+            "description": "List of tests specific to this library.",
```
Comment (beraldoleal) [inline schemas/autils.schema:52]: Will this list support expansion or regex?

File: .github/workflows/modules-tests.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+
+      - name: Install Avocado to run tests
+        run: pip3 install git+https://github.com/avocado-framework/avocado@7998c0a6d1bdd8ce60235cd4d2aa04afbebfe00e#egg=avocado_framework
+
+      - name: Run the ar module test
+        run: ./tests/test-module.py metadata/autils/archive/ar.yml
```
Comment (beraldoleal) [inline .github/workflows/modules-tests.yml:18]: Can't we make this generic enough so we don´t need to change this file for new libraries?

File: schemas/autils.schema (hunk 1 above)
Comment (clebergnu) [reply schemas/autils.schema:52]: Good question, and good idea.  We may add that when needed.  For now, I'd omit any promises that would involve extra code "just because".

File: .github/workflows/modules-tests.yml (hunk 2 above)
Comment (clebergnu) [reply .github/workflows/modules-tests.yml:18]: That one line (alone), no.  The reason is that the current form of the `test-module.py` script reads one metadata file only.

And while it'd be trivial to make `test-module.py` support multiple metadata files, I'm still trying to come up with a good enough structure for the job/suites/results.  One possibility is one `Job` per metadata file, each with many suites to match the different supported platforms.  The results produced would then either be parsed into a nice user-consumable web page -- or maybe even generated as part of an actual result plugin.

Long story short, I guess I'm trying not to commit to a behavior/structure that will definitely change soon.

Comment (beraldoleal) [reply .github/workflows/modules-tests.yml:18]: I got the idea, thanks. I would love to see this generic enough to avoid editing the file. But fair enough, let's improve after.

**Review by beraldoleal — APPROVED** [review]
